docs/technical-design/endpoints.md

```diff
@@ -0,0 +1,213 @@
+
+_**Go to `/app-api/src/resolvers/configureResolvers.ts`**_
+
+We'll import the resolver we just created.
+
+```typescript
+import { dataExportResolver } from './dataExport'
+```
+
+And also add it to the Resolvers object. We've made a query, so this line will be added to the query methods.
+
+```typescript
+dataExport: dataExportResolver(store),
+```
+
+### Generating code
+
+Once we have our backend code in place, we need to run a couple of commands to generate glue types and methods.
+
+_**On the command line, navigate to `/services/app-graphql`.**_  
+_**Run `yarn gqlgen`**_
+
+_**Again on the command line, navigate to `/services/app-api`**_  
+_**Run `yarn prisma generate`**_
+
+### Getting data on the front end
+
+One result of the `generate` commands is to create a hook that we can use on the front end to call our new resolver. The name of the hook will be of the form _useWhatYourResolverIsCalledQuery_ (it will end with "Mutation" if that's what you've created). You can pull it in and get data.
+
+```typescript
+import { useDataExportQuery } from '../../gen/gqlClient'
+const { data, loading, error } = useDataExportQuery()
+```
+
+## Lambda
+
+### Serverless configuration
+
+When we create a handler, we're telling serverless to create a new Lambda function for us in AWS. If we configure it correctly, that function will have access to things like our postgres database, which itself is hosted on AWS. Here's one existing handler configuration, found in `/services/app-api/serverless.yml`.
+
+```yaml
+reports:
+    handler: src/handlers/reports.main
+    events:
+        - http:
+              path: reports
+              method: get
+              cors: true
+              authorizer: aws_iam
+    layers:
+        - !Ref PrismaClientEngineLambdaLayer
+        - arn:aws:lambda:us-east-1:901920570463:layer:aws-otel-nodejs-amd64-ver-1-2-0:1
+    vpc:
+        securityGroupIds:
+            - ${self:custom.sgId}
+        subnetIds: ${self:custom.privateSubnets}
+```
+
+The code that runs when this handler is called is in `/app-api/src/handlers/reports.ts`.
+
+The handler points to `/app-api/src/handlers/reports.main`. We're telling the handler where to look for the file that contains the code that will run in the lambda (`src/handlers`) and also the name of the function (`main`). In `reports.ts` the exported handler is called `main`. The location and exported method of `reports.ts` have to match what's specified on the `handler:` line in our YAML file.
+
+We also provide a `path`. This handler will run when someone navigates to the site's base name, slash reports. For prod, it would be `https://mc-review.onemac.cms.gov/reports`.
+
+The `authorizer` line tells AWS that this method can only be invoked by a user who is logged in and has permissions set in AWS IAM. This will typically flow from someone being a state or CMS user, with permission to interact with the main site.
```

Comment:
How or where do we specify which type of user has access to this endpoint? In the reports endpoint I believe it is restricted to CMS users?